File: guests/python/panels.py
"""The panels conformance scene (tools/scenes/panels.steps): the inspector
is one create_window scope, its veto class one handler."""

import sys

import kaya

app = kaya.App()

with app.window(title="panels"):
    status = kaya.signal("two panels")
    with kaya.column():
        kaya.label(bind=status)  # label#0

INSPECTOR = 1


def close_asked():
    # Bound at the inspector's declaration: only THIS window's close.
    status.set("close requested")
    kaya.destroy_window(INSPECTOR)


with app.create_window(INSPECTOR, title="inspector", width=480, height=320,
                    veto_close=True, on_close_requested=close_asked):
    caption = kaya.signal("inspector pane")
    with kaya.column():
        kaya.label(bind=caption)  # label#1


sys.exit(app.run())

File: guests/js/panels.ts
// The panels conformance scene (tools/scenes/panels.steps).

import * as kaya from "kaya-gui";

const app = new kaya.App();

const { status } = app.window({ title: "panels" }, () => {
  const status = kaya.signal("two panels");
  kaya.column(() => {
    kaya.label({ bind: status }); // label#0
  });
  return { status };
});

const INSPECTOR = 1;

function closeAsked(): void {
  // Bound at the inspector's declaration: only THIS window's close.
  status.set("close requested");
  kaya.destroyWindow(INSPECTOR);
}

app.createWindow(INSPECTOR, { title: "inspector", width: 480, height: 320, vetoClose: true, onCloseRequested: closeAsked }, () => {
  const caption = kaya.signal("inspector pane");
  kaya.column(() => {
    kaya.label({ bind: caption }); // label#1
  });
});

app.run();
